app/lib/service/openid/openid_models.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'openid_models.g.dart';
+
+/// The combined data from the OpenID provider, including their signing keys.
+@JsonSerializable(includeIfNull: false, explicitToJson: true)
+class OpenIdData {
+  final OpenIdProvider provider;
+  final JsonWebKeyList jwks;
+
+  OpenIdData({
+    required this.provider,
+    required this.jwks,
+  });
+
+  factory OpenIdData.fromJson(Map<String, dynamic> json) =>
+      _$OpenIdDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$OpenIdDataToJson(this);
+}
+
+/// The (partial) data structure of the `.well-known/openid-configuration` endpoint.
+///
+/// https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata
+///
+/// TODO: add all fields required for token verification, as described in:
+///       https://github.com/dart-lang/pub-dev/pull/5795#discussion_r884877050
```

Comment:
Atleast the fields provided by github :D

Reply:
Ack. I'll add them as part of a follow-up PR.